discovery_engine_core/core/src/storage/sqlite.rs

```diff
@@ -954,6 +829,32 @@ mod tests {
         assert!(storage.search().clear().await.unwrap());
     }
 
+    #[tokio::test]
+    async fn test_empty_search() {
+        let storage = SqliteStorage::connect("sqlite::memory:").await.unwrap();
+        storage.init_database().await.unwrap();
+
+        let new_search = Search {
+            search_by: SearchBy::Query,
+            search_term: "term".to_string(),
+            paging: Paging {
+                size: 100,
+                next_page: 2,
+            },
+        };
+        storage
+            .search()
+            .store_new_search(&new_search, &[])
+            .await
+            .unwrap();
+
+        let search = storage.search().fetch().await.unwrap();
+        assert!(search.1.is_empty());
+
+        let a_search_was_closed = storage.search().clear().await.unwrap();
+        assert!(a_search_was_closed);
```

Comment:
:art:
```suggestion
        assert!(storage.search().fetch().await.unwrap().1.is_empty());
        assert!(storage.search().clear().await.unwrap());
```

Reply:
I would say assigning it to a temporary variable conveys the meaning of what is changed better.

Reply:
:rocket:  (changed it anyway)